tallycash/extension
# 🛢 Transaction nonce and gas fixes

This PR is touching multiple issues but they are tightly connected with each other

### What

**Nonce issue**
Ref #2127, #2106 - not closing for now
- Nonce should be released if the transaction failed to broadcast.
This will not fix the problem for users that already have pending or dropped transactions and nonce is in a corrupted state but it will prevent the same issue in the future.

**Gas settings issue**
Ref https://github.com/tallycash/extension/issues/2133 - 90% solved, I see the problem with custom settings - they are losing value, gas limit is ignored.
- We had a problem that the gas settings that the user was changing were not reflected in the transaction. That was because `isEIP1559EnrichedTransactionSignatureRequest(partialRequest)` was expecting `partialRequest` to have `          maxFeePerGas` and `maxPriorityFeePerGas` already set which was never true as this was the function that was supposed to set these values.

**Estimated gwei value issue**
Resolves https://github.com/tallycash/extension/issues/2112 
- Fixed the problem with incorrect gas price calculation - now Optimism and L1 gas calculation logic are not mixed up

### Testing
- setup transaction on any network (all networks should be tested)
- check Estimated network fee button value, compare with current gas prices on given network - they should be similar
- open gas settings panel, change selected option and save
- Estimated network fee should change according to selected gas setting
- set custom, very low gas fee in settings
- sign tx - it should fail to broadcast
- setup another tx, change gas settings and remember what you selected
- sign tx, check it on the scanner website - priority fee and other gas values should be set to the values you selected
- check tx nonce and previous tx nonce - they should be `N` and `N - 1`
